firefox-ios/Client/Frontend/Home/Homepage Rebuild/MessageCard/HomepageMessageCardCell.swift

```diff
@@ -0,0 +1,243 @@
+        static let cornerRadius: CGFloat = 12
+        static let dismissButtonSize = CGSize(width: 16, height: 16)
+        static let dismissButtonSpacing: CGFloat = 12
+        static let standardSpacing: CGFloat = 16
+        static let topCardSafeSpace: CGFloat = 16
+        static let bottomCardSafeSpace: CGFloat = 32
+
+        // Shadow
+        static let shadowRadius: CGFloat = 4
+        static let shadowOffset = CGSize(width: 0, height: 2)
+        static let shadowOpacity: Float = 1 // shadow opacity set to 0.16 through shadowDefault themed color
+    }
+
+    // MARK: - Properties
+    private var kvoToken: NSKeyValueObservation?
+
+    // MARK: - UI
+
+    private lazy var titleContainerView: UIView = .build { view in
+        view.backgroundColor = .clear
+    }
+
+    private lazy var bannerTitle: UILabel = .build { label in
+        label.numberOfLines = 0
+        label.lineBreakMode = .byWordWrapping
+        label.font = FXFontStyles.Regular.headline.scaledFont()
+        label.adjustsFontForContentSizeCategory = true
+        label.accessibilityIdentifier = a11y.titleLabel
+    }
+
+    private lazy var descriptionText: UILabel = .build { label in
+        label.numberOfLines = 0
+        label.lineBreakMode = .byWordWrapping
+        label.font = FXFontStyles.Regular.subheadline.scaledFont()
+        label.adjustsFontForContentSizeCategory = true
+        label.accessibilityIdentifier = a11y.descriptionLabel
+    }
+
+    private lazy var ctaButton: PrimaryRoundedButton = .build { button in
+        button.addTarget(self, action: #selector(self.handleCTA), for: .touchUpInside)
+    }
+
+    private lazy var dismissButton: UIButton = .build { [weak self] button in
+        button.setImage(
+            UIImage(named: StandardImageIdentifiers.Large.cross)?.withRenderingMode(.alwaysTemplate),
+            for: .normal
+        )
+        button.addTarget(self, action: #selector(self?.dismissCard), for: .touchUpInside)
+        button.accessibilityLabel = BannerCopy.HomeTabBannerCloseAccessibility
+    }
+
+    private lazy var textStackView: UIStackView = .build { stackView in
+        stackView.axis = .vertical
+        stackView.alignment = .fill
+        stackView.distribution = .fill
+        stackView.spacing = UX.textSpacing
+    }
+
+    private lazy var cardView: UIView = .build { view in
+        view.layer.cornerRadius = UX.cornerRadius
+    }
+
+    // MARK: - Inits
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        setupLayout()
+        observeCardViewBounds()
+    }
+
+    required init(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    deinit {
+        kvoToken?.invalidate()
+    }
+
+    func configure(state: MessageCardState, theme: Theme) {
+        applyGleanMessage(with: state.title, description: state.description, buttonLabel: state.buttonLabel)
+        applyTheme(theme: theme)
+        ctaButton.applyTheme(theme: theme)
+    }
```

Comment:
copied most of this class over, mainly changed this method from the original in `LegacyHomepageMessageCardCell`

Previous implementation
```
func configure(viewModel: HomepageMessageCardViewModel, theme: Theme) {
        self.viewModel = viewModel

        if let message = viewModel.getMessage(for: .newTabCard) {
            applyGleanMessage(message)
        }

        applyTheme(theme: theme)
        ctaButton.applyTheme(theme: theme)
    }
    ```
    Also moved the shadow constants here instead of calling it from HomepageViewModel